MudCircularProgress: Add ChildContent to Show Centered Content in the Progress

mckaragoz opened the pull request:
<!--
MAKE SURE TO READ THE CONTRIBUTING GUIDE BEFORE CREATING A PR
https://github.com/MudBlazor/MudBlazor/blob/dev/CONTRIBUTING.md

Testing sections can be removed for documentation changes
-->

<!-- Provide a general summary of your changes in the Title above -->
<!-- Keep the title short and descriptive, as it will be used as a commit message -->

## Description
This PR based on @fondraco 's PR #7419 .
Fixes #2520 , Fixes #6437 .

According to other PR, this PR is simplified. Changed `PercentageContent` name to `ChildContent` and delete math formulas. Users can use value directly inside of ChildContent. Also deleted `ShowPercentage` parameter, because it's unnecessary, if user doesn't describe ChildContent there will be no issue. And the old PR's formulas don't support double value's display, this PR supports maximum flexibility.


https://github.com/user-attachments/assets/664e1369-b249-4745-82e9-9e703d9c056b

As the same with other PR, there is a special situation with `Indetermine` circular progress, because it also rotates the child content. So we didn't touch indetermine state. It's always possible to center some other content via regular CSS. So we only show an example in the docs and guide users how to achieve centered content:

![Ekran görüntüsü 2025-05-01 181459](https://github.com/user-attachments/assets/d5519ef9-c4f5-4096-8367-12625a7ab3a2)


## How Has This Been Tested?
<!-- All PRs should implement unit tests if possible. -->
<!-- Please describe how you tested your changes. -->
<!-- Have you created new tests or updated existing ones? -->
<!-- e.g. unit | visually | none -->

## Type of Changes
<!-- What type of changes does your code introduce? Put an `x` in only one box that applies best: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (fix or improvement to the website or code docs)

<!-- If you made any visual changes, provide screenshots of before/after. If it has moving parts, please attach a high quality video. -->

## Checklist
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] The PR is submitted to the correct branch (`dev`).
- [x] My code follows the code style of this project.
- [ ] I've added relevant tests.

mckaragoz committed:
MudCircularProgress: Add ChildContent to Show Centered Content in the Progress

henon commented:
Nice. Is this issue (https://github.com/MudBlazor/MudBlazor/pull/7419#issuecomment-2023196503) also fixed?

Garderoben commented on src/MudBlazor.Docs/Pages/Components/Progress/Examples/ProgressCircularInterminateExample.razor:
We should avoid using style in our examples since its bad practice.

mckaragoz replied:
Understand. Do you prefer remove all or change this with `<style>` brackets?

Garderoben replied:
Just want to see if its possible it any other way, sometimes its no way around using the style attribute.

But can you move this to its own example? I feel like your description of the issue here on GitHub why its done like this for intermediate is good info. "Intermediate with content" and separating it.

mckaragoz commented:
> Nice. Is this issue ([#7419 (comment)](https://github.com/MudBlazor/MudBlazor/pull/7419#issuecomment-2023196503)) also fixed?

Probably yes, because i removed all built-in design included `PercentageContent` with a predefined Typo. With this PR, user only has `ChildContent` (an empty container) so they will choose what they want to place inside the container. If the content will be too big, it will be user's fault.

mckaragoz committed:
Add Indeterminate State ChildContent

mckaragoz commented:
@Garderoben @henon awesome news. When i added reversed animation inside the indeterminate circular's child content, it looks perfectly. So we have a perfect child content now. Also arranged docs.


https://github.com/user-attachments/assets/c34e3e1a-0ef9-4f49-8406-ed8eab555e21

henon commented on src/MudBlazor.Docs/Pages/Components/Progress/ProgressPage.razor:
```diff
@@ -51,6 +51,15 @@
 
+                <DocsPageSection>
+                    <SectionHeader Title="Percentage Content">
+                        <Description>The desired elements can be placed in the component with the <CodeInline>ChildContent</CodeInline> parameter.</Description>
```
`Title="Custom Content"`


`You can place any content in the middle of the circular progress using the <CodeInline>ChildContent</CodeInline> parameter.`

henon reviewed (approved):
Just a little suggestion for the documentation. Otherweise looks good from my perspective.

ScarletKuro committed:
Update ProgressCircularPercentageExample.razor

ScarletKuro committed:
Update MudProgressCircular.razor.cs

ScarletKuro commented on src/MudBlazor/Components/Progress/MudProgressCircular.razor.cs:
```diff
@@ -17,6 +17,7 @@ namespace MudBlazor
     public partial class MudProgressCircular : MudComponentBase
     {
         private int _svgValue;
+        private double _fraction;
```
I actually don't see a reason why this was moved from being local variable of `ToSvgValue` to here?
I don't see it being used anywhere else.

mckaragoz replied:
It was old code that fondraco added. We can delete it if unnecessary.

ScarletKuro commented:
Also the `DefaultStructure` test is failing.

mckaragoz commented:
> Also the `DefaultStructure` test is failing.

Yes i also want to ask it, what is this test?

mckaragoz committed:
Docs Description Change

mckaragoz committed:
Changed Frantion field to old state

mckaragoz commented:
PR is ready, but no idea why tests failed.

Garderoben commented:
> PR is ready. Saw the failed test, just need the increase child item count. But this was my old branch, have some other problems with test project, i can't run. Please ignore the test for now and merge this PR, i will look at the test with another PR immediately.

Approved, but  build/test still fails?

mckaragoz commented:
> You can make the necessary branch changes from github UI, especially if it's "just need the increase child item count".
> I don't think it's wise to accept PR if the tests stops to pass for everyone

There are more than one child content check in this test, i don't know which one I should change exactly. As i said this branch is in my old hard drive, can't build it successfully. If you merge this, i can make a test fix in the same day, other PR's won't affect.

By the way 4 more tests failing for me (date picker) probably for culture issues. I will fix this test with these ones in one PR.

ScarletKuro commented:
> If you merge this, i can make a test fix in the same day, other PR's won't affect.

We can wait when you fix this the same day. This PR is not urgent.

> As i said this branch is in my old hard drive, can't build it successfully

But this is Git, you can fetch your branch, everything is on the server. What’s stopping you from getting back to your work on the new hard drive / PC?

```bash
git clone https://github.com/mckaragoz/MudBlazor.git
cd MudBlazor
git fetch --all
git branch -r   # (Optional, just to confirm the branch exists)
git checkout CircularProgressPercentage
```

Then you can continue working on it. At worst, you can download GitHub Desktop ([https://desktop.github.com/download/](https://desktop.github.com/download/)). In this PR, just click "<> Code" and then "Open with GitHub Desktop", it will handle most of the setup for you.

I’m not trying to be difficult, I just want people to learn the tools and how to use them when situations like this come up.

mckaragoz committed:
Fix Test

mckaragoz commented:
> But this is Git, you can fetch your branch, everything is on the server. What’s stopping you from getting back to your work on the new hard drive / PC?

Oh yes, it's my delusion. Then ready now.
